main.tf

```diff
@@ -161,7 +161,7 @@ resource "google_alloydb_instance" "primary" {
       dynamic "authorized_external_networks" {
```

Comment:
@rayanebel I think we have to change this block and make it similar to the  block you added for read instances. 

```
  dynamic "network_config" {
    for_each = var.primary_instance.enable_public_ip ? ["network_config"] : []
    content {
      enable_public_ip = var.primary_instance.enable_public_ip
      dynamic "authorized_external_networks" {
        for_each = var.primary_instance.cidr_range == null ? [] : toset(var.primary_instance.cidr_range)
        content {
          cidr_range = authorized_external_networks.value
        }
      }
    }
  }
```

Reply:
@imrannayer Done, I've tested in my cluster and its working